src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2197,6 +2203,23 @@ def aead_cipher_supported(self, cipher):
             self._lib.EVP_get_cipherbyname(cipher_name) != self._ffi.NULL
         )
 
+    @contextlib.contextmanager
+    def _zeroed_bytearray(self, length):
+        """
+        This method creates a bytearray, which we copy data into (hopefully
+        also from a mutable buffer that can be dynamically erased!), and then
+        zero when we're done.
+        """
+        ba = bytearray(length)
+        try:
+            yield ba
+        finally:
+            # We zero the bytearray this way because at the moment we're not
+            # sure of a better way that can guarantee it overwrites the
+            # memory and doesn't just replace the underlying char *.
+            for i in range(length):
```

Comment:
Please factor this into its own function and use it in the other place that we use `memmove` to zero

Reply:
We're doing `memmove` in the other place on a `char[]` so `memmove` is the right thing to use there.

Reply:
A loop also works though.